assembly/src/parsers/field_ops.rs

```diff
@@ -373,6 +384,84 @@ pub fn parse_gte(span_ops: &mut Vec<Operation>, op: &Token) -> Result<(), Assemb
     Ok(())
 }
 
+// POWER OF TWO HELPER FUNCTIONS
+// ================================================================================================
+
+/// Extract the least significant bit of the top element iteratively and performs
+/// power of 2 operation on the individual bit. These individual powers
+/// are combined later to calculate the power of 2 on the top value of the stack.
+///
+/// The expected starting state of the stack (from the top) is: [a, ...].
+///
+/// After these operations, the stack state will be: [2^a, ...].
+fn aggregate_power_2(span_ops: &mut Vec<Operation>, unsafe_mode: bool) {
+    const MOST_SIGNIFICANT_BIT: u32 = 5;
+
+    // `safe` Mode
+    if !unsafe_mode {
+        // Checks if the top element of the stack is less than 64 or not.
+        span_ops.push(Operation::Dup0);
+        span_ops.push(Operation::Push(Felt::new(64)));
+        check_lt(span_ops);
+        span_ops.push(Operation::Assert);
```

Comment:
I think this can be done in 5 cycles as:
```
push.64
u32div.unsafe
swap
eqz
assert
```

Reply:
Yes, done!